include/alp/algorithms/householder_qr.hpp

```diff
@@ -78,31 +83,31 @@ namespace alp {
 			auto Qdiag = alp::get_view< alp::view::diagonal >( Q );
 			rc = rc ? rc : alp::set( Qdiag, one );
 			if( rc != SUCCESS ) {
-				std::cerr << " set( Q, I ) failed\n";
+				std::cerr << " alp::set( Q, I ) failed\n";
 				return rc;
 			}
 
 			// Out of place specification of the computation
-			Matrix< D, GeneralType, Dense > RR( n, m );
+			Matrix< D, GeneralType, alp::Dense, GenView, GenImfR, GenImfC > RR( n, m );
```

Comment:
GenView, ImfR and ImfC should be used in input parameters to support passing matrix views to the function. For any internal variable within the function these should not be used. In summary, just remove them here.